sys/include/ztimer64.h

```diff
@@ -0,0 +1,519 @@
+ * @param[in]   timeout         relative timeout, in @p clock time units
+ *
+ * @return  >=0 if a message was received
+ * @return  -ETIME on timeout
+ */
+static inline int ztimer64_msg_receive_timeout(ztimer64_clock_t *clock,
+                                               msg_t *msg,
+                                               uint64_t timeout)
+{
+    return ztimer64_msg_receive_until(clock, msg,
+                                      ztimer64_offset2absolute(clock, timeout));
+}
+
+#define MSG_ZTIMER64 0xc83f   /**< msg type used by ztimer64_msg_receive_timeout */
+
+/**
+ * @brief Suspend the calling thread until the time (@p last_wakeup + @p period)
+ *
+ * This function can be used to create periodic wakeups.
+ *
+ * When the function returns, @p last_wakeup is set to
+ * (@p last_wakeup + @p period).
+ *
+ * @c last_wakeup should be set to ztimer64_now(@p clock) before first call of the
+ * function.
+ *
+ * If the time (@p last_wakeup + @p period) has already passed, the function
+ * sets @p last_wakeup to @p last_wakeup + @p period and returns immediately.
+ *
+ * @note if you period is smaller than 2^32 ticks, consider using the 32bit version.
+ *
+ * @param[in]   clock           ztimer clock to operate on
+ * @param[in]   last_wakeup     base time stamp for the wakeup
+ * @param[in]   period          time in ticks that will be added to @p last_wakeup
+ */
+void ztimer64_periodic_wakeup(ztimer64_clock_t *clock, uint64_t *last_wakeup,
+                              uint64_t period);
+
+/**
+ * @brief   Put the calling thread to sleep until the specified time
+ *
+ * @param[in]   clock           ztimer clock to use
+ * @param[in]   target          wakeup time, in @p clock time
+ */
+void ztimer64_sleep_until(ztimer64_clock_t *clock, uint64_t target);
+
+/**
+ * @brief   Put the calling thread to sleep for the specified number of ticks
+ *
+ * @param[in]   clock           ztimer clock to use
+ * @param[in]   duration        duration of sleep, in @p ztimer time units
+ */
+static inline void ztimer64_sleep(ztimer64_clock_t *clock, uint64_t duration)
+{
+    ztimer64_sleep_until(clock, ztimer64_offset2absolute(clock, duration));
+}
+
+/**
+ * @brief   Busy-wait until specified target time
+ *
+ * @note: This blocks lower priority threads. Use only for *very* short delays.
+ *
+ * @param[in]   clock           ztimer clock to use
+ * @param[in]   target          time when spinning should end, in @p clock time
+ */
+static inline void ztimer64_spin_until(ztimer64_clock_t *clock, uint64_t target)
+{
+    while (ztimer64_now(clock) <= target) {}
+}
+
+/**
+ * @brief   Busy-wait specified duration
+ *
+ * @note: This blocks lower priority threads. Use only for *very* short delays.
+ *
+ * @param[in]   clock           ztimer clock to use
+ * @param[in]   duration        duration to spin, in @p clock time units
+ */
+static inline void ztimer64_spin(ztimer64_clock_t *clock, uint64_t duration)
```

Comment:
Same as above

Reply:
this I'll remove, there's no need to allow this over ztimer32_spin()